packages/bsky/src/hydration/hydrator.ts

```diff
@@ -104,6 +104,10 @@ export class Hydrator {
       listUris.push(...listUrisFromProfileViewer(item))
     })
     const listState = await this.hydrateListsBasic(listUris, viewer)
+    // if a list no longer exists or is not a mod list, then remove from viewer state
+    profileViewers?.forEach((item) => {
+      removeNonModListsFromProfileViewer(item, listState)
+    })
```

Comment:
I'm a little torn on modifying the `ProfileViewer`s in this way.  There's an implicit dependency between `getProfileViewerStates()` and this logic.  We could also handle it on the flip side, when blocks are being evaluated, in `viewerBlockExists()` and `viewerMuteExists()`, and in some ways I think that fits better into how hydration vs. view logic currently works.  Either case introduces some dependencies, so I'd be for leaving another comment or two to document those.  That would be either:
 - documenting in `getProfileViewerStates()` the dependence on a later call to `removeNonModListsFromProfileViewer()` for evaluation of blocks/mutes or
 - documenting in `viewerMute/BlockExists()` that referenced lists must be hydrated.

Reply:
yeah i thought about doing it in the view layer

but I think there's a decent argument that this should be handled in hydration. in v1, we were doing this in hydration & the lists on ProfileViewers don't actually "exist". Imo this isn't "view logic", it's an additional piece of logic on hydration that would ideally be handled in the hydration query but isn't because everything is denormalized & this is expensive to materialize 🤔 

Def down to rework logic/document it better

Reply:
I agree it would be more ideal for this to be in `getProfileViewerStates` but then it mixes it up with list logic which is kinda pretty strange because we have to jam a list hydration call in there...

maybe we could add a higher level `getProfilerViewerStates` to the hydrator that handles this logic

and have a lower level `getUncheckedProfileViewerStates` on the profile hydrator that calls out the response is incomplete